src/Catalog/Monitoring/ValidationTestCollector.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Result;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test;
+using NuGet.Services.Metadata.Catalog.Persistence;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// Runs a <see cref="PackageValidationTestRunner"/> against collected packages.
+    /// </summary>
+    public class ValidationTestCollector : SortingIdVersionCollector
+    {
+        public ValidationTestCollector(Storage auditingStorage, PackageValidationTestRunner packageValidationTestRunner, Uri index, Func<HttpMessageHandler> handlerFunc = null, ILogger logger = null) : base(index, handlerFunc)
+        {
+            _auditingStorage = auditingStorage;
+            _packageValidationTestRunner = packageValidationTestRunner;
+            _logger = logger;
+        }
+        
+        public IEnumerable<PackageValidationTestResult> PackageValidationResults => new List<PackageValidationTestResult>();
```

Comment:
in general, let's only add code when we need it. that way if plans change in the future there isn't hard to find dead code.